### PR TITLE
improved logging of objectore errors

### PR DESCRIPTION
**Pull request by icewind1991**
Log the full exception, urn and path on object store errors

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
Tested and works

**Comment by MorrisJobke** [comment]
Could you backport this to stable12 please? Thanks

cc @karlitschek

**Comment by karlitschek** [comment]
definitely :+1: :-)

**Review by nickvergessen — APPROVED** [review]

File: lib/private/Files/ObjectStore/ObjectStoreStorage.php [hunk 1]
```diff
@@ -185,7 +189,8 @@ public function unlink($path) {
 				$this->objectStore->deleteObject($this->getURN($stat['fileid']));
 			} catch (\Exception $ex) {
 				if ($ex->getCode() !== 404) {
-					\OCP\Util::writeLog('objectstore', 'Could not delete object: ' . $ex->getMessage(), \OCP\Util::ERROR);
+					$this->logger->error('Could not delete object ' . $this->getURN($stat['fileid']) . ' for ' . $path, ['app' => 'objectstore']);
```
Comment (nickvergessen) [inline lib/private/Files/ObjectStore/ObjectStoreStorage.php]: Use https://github.com/nextcloud/server/blob/43cfad705cddb57c6a79dc51a12d151e14533064/lib/public/ILogger.php#L130-L136 instead?

Same below

Comment (MorrisJobke) [inline lib/private/Files/ObjectStore/ObjectStoreStorage.php]: Fixed

**Pull request merged**